### PR TITLE
SPM-1132: Return 400 when filter is not supported

### PR DESCRIPTION
**Pull request by AlesKas**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: manager/controllers/advisory_detail.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ func AdvisoryDetailHandler(c *gin.Context) {
 		return
 	}
 
+	if filterInURL(c.Request.URL.String()) {
```
Comment (Josca) [inline manager/controllers/advisory_detail.go:64]: I would rather use `isFilterInURLValid(c *gin.Context) bool`. When it returns "false" we can just call return:
~~~golang
if !isFilterInURLValid(c) {
   return
}
~~~
You can get url string from c and also log error and write status code accordingly directly in the method so we don't need to repeat it in all method callings.

File: manager/controllers/advisory_detail_test.go [hunk 2]
```diff
@@ -122,3 +123,18 @@ func TestAdvisoryDetailCachePreloading(t *testing.T) {
 	assert.True(t, ok)
 	advisoryDetailCache.Purge()
 }
+
+// nolint: errcheck
```
Comment (Josca) [inline manager/controllers/advisory_detail_test.go:127]: I think you can use:
~~~golang
var errResp utils.ErrorResponse
ParseReponseBody(t, w.Body.Bytes(), &errResp)
assert.Equal(t, InvalidOffsetMsg, errResp.Error)
~~~
... as used elsewhere so you don't need to use this "nolint" annotation.

File: manager/controllers/utils.go [hunk 3]
```diff
@@ -81,7 +82,6 @@ func ApplySort(c *gin.Context, tx *gorm.DB, fieldExprs database.AttrMap,
 
 func validateFilters(q QueryMap, allowedFields database.AttrMap) error {
 	for key := range q {
-		// system_profile is hadled by tags, so it can be skipped
```
Comment (Josca) [inline manager/controllers/utils.go:84]: Why do you remove this comment?

**Review by Josca — CHANGES_REQUESTED** [review]
1) I've added some ideas.
2) There is a typo in the commit message.
3) Please include SPM-XYZ to both commit message and pr title.

File: manager/controllers/utils.go (hunk 3 above)
Comment (AlesKas) [reply manager/controllers/utils.go:84]: my bad, I must have somehow delete without noticing it, returning it

**Review by Josca — APPROVED** [review]

**Pull request merged**